src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -94,13 +94,9 @@ public async Task SpecializeAsync()
             if (_workerRuntime != null && languageWorkerChannel != null)
             {
                 _logger.LogInformation("Loading environment variables for runtime: {runtime}", _workerRuntime);
-                IObservable<WorkerProcessReadyEvent> processReadyEvents = _eventManager.OfType<WorkerProcessReadyEvent>()
-                .Where(msg => string.Equals(msg.Language, _workerRuntime, StringComparison.OrdinalIgnoreCase))
-                .Timeout(workerInitTimeout);
-                languageWorkerChannel.SendFunctionEnvironmentReloadRequest();
-
-                // Wait for response from language worker process
-                await processReadyEvents.FirstAsync();
+                TaskCompletionSource<bool> reloadTask = new TaskCompletionSource<bool>();
+                languageWorkerChannel.SendFunctionEnvironmentReloadRequest(reloadTask);
```

Comment:
Why is this being passed from the consumer? any reason why this token/task isn't fully managed by the channel instead?

Reply:
SendFunctionEnvironmentReloadRequest is called during specialization. Caller of the method needs to wait until FunctionEnvironmentReloadResponse is received from the language worker. Let me know if there is a better way to achieve this.

Reply:
This shouldn't be manged by the consumer. Also left a comment in the method implementation as the new approach is a bit fragile and opens up the opportunity for other potential issues. This should be managed by the worker channel